tests/suite/upgrade_test.go

```diff
@@ -0,0 +1,246 @@
+package suite
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
```

Comment:
It would be nice if the test suite didn't have to know about vegeta, in case we need to swap it out for a different load-testing tool in the future.

Reply:
Added some wrappers, let me know what you think.